libs/autofile/autofile.go

```diff
@@ -79,7 +79,7 @@ func OpenAutoFile(path string) (*AutoFile, error) {
 	signal.Notify(af.hupc, syscall.SIGHUP)
 	go func() {
 		for range af.hupc {
-			af.closeFile()
+			af.closeFile() //nolint:errcheck // ignore error
```

Comment:
Another trick here is to do `_ = af.closeFile()`, which is idiomatic for "yes, I know this returns an error, and I've explicitly chosen to ignore it."